cmd/snapshot_export.go

```diff
@@ -17,38 +28,172 @@ func (o *snapshotExportOutput) toJSON()  { outputJSON(o) }
 func (o *snapshotExportOutput) toText()  { outputText(o) }
 func (o *snapshotExportOutput) toTable() { outputTable(o) }
 
-func init() {
-	snapshotCmd.AddCommand(&cobra.Command{
-		Use:   "export <snapshot ID>",
-		Short: "export snapshot",
-		Long: fmt.Sprintf(`This command exports a volume snapshot.
+var snapshotExportCmd = &cobra.Command{
+	Use:   "export <snapshot ID>",
+	Short: "export snapshot",
+	Long: fmt.Sprintf(`This command exports a volume snapshot.
 
 Supported output template annotations: %s`,
-			strings.Join(outputterTemplateAnnotations(&snapshotExportOutput{}), ", ")),
-		RunE: func(cmd *cobra.Command, args []string) error {
-			if len(args) != 1 {
-				return cmd.Usage()
-			}
-			return output(exportSnapshot(args[0]))
-		},
-	})
+		strings.Join(outputterTemplateAnnotations(&snapshotExportOutput{}), ", ")),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 1 {
+			return cmd.Usage()
+		}
+
+		filePath, err := cmd.Flags().GetString("download")
+		if err != nil {
+			return err
+		}
+
+		snapshot, err := exportSnapshot(args[0])
+		if err != nil {
+			return err
+		}
+
+		if !cmd.Flags().Changed("download") {
+			return output(&snapshotExportOutput{
+				URL:      snapshot.PresignedURL,
+				Checksum: snapshot.MD5sum,
+			}, nil)
+		}
+
+		if err = downloadExportedSnapshot(filePath, args[0], snapshot.PresignedURL, snapshot.MD5sum); err != nil {
+			return err
+		}
```

Comment:
`downloadExportedSnapshot` should do only what its name implies: download the template:

```suggestion
		if err = downloadExportedSnapshot(filePath, args[0], snapshot.PresignedURL); err != nil {
			return err
		}
```

Then, if no error occurred during template download you can call `checkExportedSnapshot`:

```go
		if err = checkExportedSnapshot(filePath, snapshot.MD5sum); err != nil {
			return err
		}
```